### PR TITLE
Change  to firebase functions

### PR DESCRIPTION
**Pull request by king-11**
- updated workflows and directory structure

### CODE REVIEW

File: functions/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        names?.map((name) =>
+          ctx.reply(`Hey ${name}! I'm really interested in you, so can you please introduce yourself?`),
+        ),
+      );
+    }
+  },
+);
+
+bot.on(
+  'left_chat_member',
```
Comment (shivanshs9) [inline functions/src/index.ts:36]: I don't think we need Lucy to comment all the time on someone leaving.. It's not interactive.

File: .gitignore [hunk 2]
```diff
@@ -78,6 +78,19 @@ npm-debug.log*
...
+
+# Firebase cache
+.firebase/
+
+# Firebase config
+
+# Uncomment this if you'd like others to create their own Firebase project.
+# For a team working on the same Firebase project(s), it is recommended to leave
+# it commented so all members can deploy to the same project(s) in .firebaserc.
+.firebaserc
```
Comment (shivanshs9) [inline .gitignore:93]: Right now it's anyways not pushed?

**Review by shivanshs9 — CHANGES_REQUESTED** [review]
Was it really necessary to change the working directory?
Even if it was necessary, in any project `package.json` and other config files are mostly preferred to be at root-level folder.

**Comment by king-11** [comment]
> Was it really necessary to change the working directory?
> Even if it was necessary, in any project `package.json` and other config files are mostly preferred to be at root-level folder.

I did checkout but it seems though firebase functions have to be kept inside a functions folder for deployment and identification purposes as per firebase docs

File: functions/src/index.ts (hunk 1 above)
Comment (king-11) [reply functions/src/index.ts:36]: the issue abt people leaving and bot stopping so how shall I change for it

File: .gitignore (hunk 2 above)
Comment (king-11) [reply .gitignore:93]: didn't get it, the config will be created on adding firebase project by running ```firebase add --use```

Comment (shivanshs9) [reply .gitignore:93]: so you haven't yet added any firebase project with it, right?

Comment (king-11) [reply .gitignore:93]: yes

**Pull request closed without merging**